### DISCUSSION TITLE
_document 'link' element doesn't work in production.

### DISCUSSION THREAD

**Discussion by mena234** [vercel/next.js#38683] [Help] [answered]
I'm trying to get a font from [google fonts](https://fonts.googleapis.com), so I used `link` element in _document file.

`
    <Html>
      <Head>
        <link
          href='https://fonts.googleapis.com/css?family=Poppins:400,400i,700,700i|Rock+Salt:400,400i,700&subset=devanagari,latin_ext,latin&display=swap'
          rel="stylesheet"
        />
      </Head>
      <body>
        <Main />
        <NextScript />
      </body>
    </Html>
 `

The font works fine on development, and I can see in the `head` tag, the `link` element like that.

`<link href="https://fonts.googleapis.com/css?family=Poppins:400,400i,700,700i|Rock+Salt:400,400i,700&amp;subset=devanagari,latin_ext,latin&amp;display=swap" rel="stylesheet">`

But in production, it doesn't bind the `url` to `href` attribute, but it binds it with `data-href`, So it doesn't work, and it looks like that.

`<link data-href="https://fonts.googleapis.com/css?family=Poppins:400,400i,700,700i|Rock+Salt:400,400i,700&amp;subset=devanagari,latin_ext,latin&amp;display=swap" rel="stylesheet">`

Any idea what can be causing this issue.

Thanks a lot.

**Comment by icyJoseph** [comment]
This is part of an optimisation Next.js does to load fonts ASAP.

What problem are you actually seeing? Are the fonts not loading?

Refer to this page: https://nextjs.org/learn/seo/improve/fonts from the learning resources.

As usual the docs gives us some options, https://nextjs.org/docs/basic-features/font-optimization#disabling-optimization, talks about disabling this optimisation.

I imagine if the font family is not loading, then the issue might be in the CSS that's taking care of that.

**Comment by officialrajdeepsingh** [comment]
Check your CSS file and add the proper font family on the element.

**Reply by mena234** [reply]
Unfortunately, it didn't work this way too.

**Comment by icyJoseph** [comment, marked as answer]
OP, I recently learned that in next@12.1.6, pages with GetServerSideProps do not load fonts, it was a bug! It has been fixed in next@12.2.x, see https://github.com/vercel/next.js/issues/35835 for reference.
